### ISSUE TITLE
[FEATURE] Allow dynamic variants to be recalculated on every iteration when using repeat.

### ISSUE THREAD

**Issue by DanEscamilla** [framer/motion#1247]
**Is your feature request related to a problem? Please describe.**
I am trying to make a variant animation (In a child component, so it does not control the variant state) that  loops between random values. Currently it is not possibly since there is no way to change the variant for every iteration when using the repeat prop.

**Describe the solution you'd like**
A solution could be that when a dynamic variant has a repeating transition, it executes the TargetResolver for every iteration, that way you can have logic that is applied to every iteration instead of only once. Or have a prop to enable such functionality.

**Describe alternatives you've considered**
I tried to use onAnimationEnd to re-render the component and change the values on the variant but that approach does not work since the animation is only executed when entering the variant state so to speak.  And if you use the repeat prop for the variant, it does not trigger onAnimationEnd so I cannot rerender the variants. I don't think that approach would be a good idea anyways. 

I tried useAnimation but, it doesn't work since it breaks the variant propagation from the parent.

**Comment by mattgperry** [comment]
To set expectations I'm closing this as a `wontfix` as it probably won't be implemented unless we find a need for it within Framer.
